attachments_to_filesystem/models/ir_attachment.py

```diff
@@ -0,0 +1,145 @@
+            cr, uid, 'ir_attachment.location')
+        if location:
+            # we assume the user knows what she's doing. Might be file:, but
+            # also whatever other scheme shouldn't matter. We want to bring
+            # data from the database to there
+            pass
+        else:
+            ir_model_data._update(
+                cr, uid, 'ir.config_parameter', module_name,
+                {
+                    'key': 'ir_attachment.location',
+                    'value': 'file',
+                },
+                xml_id='config_parameter_ir_attachment_location',
+                context=context)
+
+        # synchronous behavior
+        if self.pool['ir.config_parameter'].get_param(
+                cr, uid, 'attachments_to_filesystem.move_during_init'):
+            self._attachments_to_filesystem_cron(cr, uid, context, limit=None)
+            return
+
+        # otherwise, configure our cronjob to run next night
+        user = self.pool['res.users'].browse(cr, uid, uid, context=context)
+        next_night = datetime.now() + relativedelta(
+            hour=01, minute=42, second=0)
+        next_night = pytz.timezone(user.tz).localize(next_night).astimezone(
+            pytz.utc).replace(tzinfo=None)
+        if next_night < datetime.now():
+            next_night += relativedelta(days=1)
+        ir_cron.write(
+            cr, uid,
+            [
+                ir_model_data.get_object_reference(
+                    cr, uid, module_name, 'cron_move_attachments')[1],
+            ],
+            {
+                'nextcall':
+                next_night.strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+                'doall': True,
+            },
+            context=context)
+
+    def _attachments_to_filesystem_cron(self, cr, uid, context=None,
+                                        limit=10000):
+        """Do the actual moving"""
+        limit = int(
+            self.pool['ir.config_parameter'].get_param(
+                cr, uid, 'attachments_to_filesystem.limit', '0')) or limit
+        ir_attachment = self.pool['ir.attachment']
+        attachment_ids = ir_attachment.search(
+            cr, uid, [('db_datas', '!=', False)], limit=limit, context=context)
+        logging.info('moving %d attachments to filestore', len(attachment_ids))
+        counter = 1
+        # attachments can be big, so we read every attachment on its own
+        for attachment_id in attachment_ids:
+            attachment_data = ir_attachment.read(
+                cr, uid, [attachment_id], ['datas', 'res_model'],
+                context=context)[0]
+            if attachment_data['res_model'] and not self.pool.get(
+                    attachment_data['res_model']):
+                logging.warning(
+                    'not moving attachment %d because it links to unknown '
+                    'model %s', attachment_id, attachment_data['res_model'])
+                continue
+            ir_attachment.write(
```

Comment:
If its going to fail it will be here.  Have you considered a try except with savepoint, so it can keep going, or at least gracefully exit if some error occurs.